### PR TITLE
ref(replays): Refactor useVirtualizedGrid() to encapsulate some more resize logic

### PR DESCRIPTION
**Pull request by ryan953**

### CODE REVIEW

File: static/app/views/replays/detail/useVirtualizedGrid.tsx [hunk 1]
```diff
@@ -1,37 +1,84 @@
...
+  /**
+   * List of other values that should trigger re-computing column sizes
+   * */
   deps: DependencyList;
-  ref: RefObject<MultiGrid>;
-  wrapperRef: RefObject<HTMLDivElement>;
+  /**
+   * There must be one column with a dynamic width, so the table can fill all available width inside the container
+   */
+  dyanmicColumnIndex: number;
```
Comment (billyvg) [inline static/app/views/replays/detail/useVirtualizedGrid.tsx:27]:
```suggestion
  dynamicColumnIndex: number;
```

File: tags.lock [hunk 2]
```diff
@@ -0,0 +1 @@
+18298
```
Comment (billyvg) [inline tags.lock:1]: what's this?

File: static/app/views/replays/detail/useVirtualizedGrid.tsx [hunk 3]
```diff
@@ -1,37 +1,84 @@
...
+            ),
+            200
+          );
+          return colWidth;
+        }
+
+        return cache.columnWidth({index});
+      },
+    [cache, columnCount, dyanmicColumnIndex, scrollBarWidth]
+  );
```
Comment (billyvg) [inline static/app/views/replays/detail/useVirtualizedGrid.tsx:75]: imo this is kinda hard to read with all the nesting

File: static/app/views/replays/detail/useVirtualizedGrid.tsx (hunk 1 above)
Comment (ryan953) [reply static/app/views/replays/detail/useVirtualizedGrid.tsx:27]: Autocomplete is amazing!

File: tags.lock (hunk 2 above)
Comment (ryan953) [reply tags.lock:1]: ugh, this pops up on my machine sometimes :(

good eye for it

**Review by billyvg — APPROVED** [review]

**Pull request merged**